python/lsst/fgcmcal/fgcmBuildStarsTable.py

```diff
@@ -0,0 +1,347 @@
+            visit = dataRef.dataId[self.config.visitDataRefName]
+
+            groupedDataRefs[visit] = []
+
+            # Find an existing calexp (we need for psf and metadata)
+            # and make the relevant dataRef
+            for ccdId in ccdIds:
+                calexpRef = butler.dataRef('calexp', dataId={self.config.visitDataRefName: visit,
+                                                             self.config.ccdDataRefName: ccdId})
+                if calexpRef.datasetExists():
+                    groupedDataRefs[visit].append(calexpRef)
+                    break
+
+            # And append this dataRef
+            groupedDataRefs[visit].append(dataRef)
+
+        return groupedDataRefs
+
+    def fgcmMakeAllStarObservations(self, groupedDataRefs, visitCat,
+                                    calibFluxApertureRadius=None,
+                                    visitCatDataRef=None,
+                                    starObsDataRef=None,
+                                    inStarObsCat=None):
+        """
+        Compile all good star observations from visits in visitCat.  Checkpoint files
+        will be stored if both visitCatDataRef and starObsDataRef are not None.
+
+        Parameters
+        ----------
+        groupedDataRefs: `dict` of `list`s
+           Lists of `lsst.daf.persistence.ButlerDataRef`, grouped by visit.
+        visitCat: `afw.table.BaseCatalog`
+           Catalog with visit data for FGCM
+        calibFluxApertureRadius: `float`, optional
+           Aperture radius for calibration flux.  Default is None.
+        visitCatDataRef: `lsst.daf.persistence.ButlerDataRef`, optional
+           Dataref to write visitCat for checkpoints
+        starObsDataRef: `lsst.daf.persistence.ButlerDataRef`, optional
+           Dataref to write the star observation catalog for checkpoints.
+        inStarObsCat: `afw.table.BaseCatalog`
+           Input (possibly incomplete) observation catalog
+
+        Returns
+        -------
+        fgcmStarObservations: `afw.table.BaseCatalog`
+           Full catalog of good observations.
+
+        Raises
+        ------
+        RuntimeError: Raised if doSubtractLocalBackground is True and
+           calibFluxApertureRadius is not set.
+        """
+        startTime = time.time()
+
+        if (visitCatDataRef is not None and starObsDataRef is None or
+           visitCatDataRef is None and starObsDataRef is not None):
+            self.log.warn("Only one of visitCatDataRef and starObsDataRef are set, so "
+                          "no checkpoint files will be persisted.")
+
+        if self.config.doSubtractLocalBackground and calibFluxApertureRadius is None:
+            raise RuntimeError("Must set calibFluxApertureRadius if doSubtractLocalBackground is True.")
+
+        # To get the correct output schema, we use the same code as fgcmBuildStarsTask
+        # We are not actually using this mapper, except to grab the outputSchema
```

Comment:
I changed "same" to "similar", and since it's four lines that return 3 variables that I need, changing to a shared method which returns a giant tuple doesn't seem to me like a huge savings.